src/modules/flow/http-server/http-server.c

```diff
@@ -0,0 +1,626 @@
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static void
+boolean_close(struct sol_flow_node *node, void *data)
+{
+    struct http_data *mdata = data;
+
+    stop_server(mdata);
+}
+
+static int
+boolean_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    int r;
+    struct http_data *mdata = data;
+    struct sol_flow_node_type_http_server_boolean_options *opts =
+        (struct sol_flow_node_type_http_server_boolean_options *)options;
+
+    r = start_server(mdata, opts->path, boolean_response_cb);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->value.b = opts->value;
+    mdata->node = node;
+
+    return 0;
+}
+
+/* ------------------------------------- string ------------------------------------------- */
+
+static int
+string_response_cb(void *data, struct sol_http_request *request)
+{
+    int r = 0;
+    uint16_t idx;
+    bool send_json = false;
+    enum sol_http_method method;
+    struct http_data *mdata = data;
+    char str[1024] = { 0 };
+    struct sol_str_slice slice;
+    struct sol_http_param_value *value;
+    struct sol_http_response response = {
+        .api_version = SOL_HTTP_RESPONSE_API_VERSION,
+        .content = SOL_BUFFER_INIT_EMPTY,
+        .param = SOL_HTTP_REQUEST_PARAM_INIT,
+        .response_code = SOL_HTTP_STATUS_OK
+    };
+
+    method = sol_http_request_get_method(request);
+    response.url = sol_http_request_get_url(request);
+
+    SOL_HTTP_PARAM_FOREACH_IDX (sol_http_request_get_params(request), value, idx) {
+        switch (value->type) {
+        case SOL_HTTP_PARAM_POST_FIELD:
+            if (streq(value->value.key_value.key, "value")) {
+                free(mdata->value.s);
+                mdata->value.s = strdup(value->value.key_value.value);
+                SOL_NULL_CHECK(mdata->value.s, -ENOMEM);
+
+                r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+                SOL_INT_CHECK(r, < 0, r);
+            }
+            break;
+        case SOL_HTTP_PARAM_HEADER:
+            if (streq(value->value.key_value.key, HTTP_HEADER_ACCEPT)) {
+                if (strstr(value->value.key_value.value, HTTP_HEADER_CONTENT_TYPE_JSON))
+                    send_json = true;
+            }
+            break;
+        default:
+            break;
+        }
+    }
+
+    if (send_json) {
+        r = snprintf(str, sizeof(str), "{\n\t\"%s\": \"%s\"\n}", mdata->path,
+            mdata->value.s);
+    } else {
+        r = snprintf(str, sizeof(str), "%s", mdata->value.s);
+    }
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    slice.data = str;
+    slice.len = strlen(str);
+
+    r = sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
+        HTTP_HEADER_CONTENT_TYPE, (send_json) ? HTTP_HEADER_CONTENT_TYPE_JSON : HTTP_HEADER_CONTENT_TYPE_TEXT));
+    SOL_INT_CHECK_GOTO(r, != true, end);
+
+    r = sol_buffer_set_slice(&response.content, slice);
```

Comment:
Since slice is only initialized to be used here, might be better to use `sol_str_slice_from_str()` inside the call: sol_buffer_set_slice(&content,sol_str_slice_from_str(str));